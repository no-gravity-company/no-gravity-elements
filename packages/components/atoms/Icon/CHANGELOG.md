# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.5](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/icon@1.1.0...@no-gravity-elements/icon@1.1.5) (2024-03-29)

**Note:** Version bump only for package @no-gravity-elements/icon

# 1.1.0 (2024-03-14)

### Features

- Add icon to button WC ([b6a75f2](https://github.com/no-gravity-company/no-gravity-elements/commit/b6a75f2d6d07caf3ec001b14ad9cc38d79b2274a))
- Icon WC ([2c7e3f0](https://github.com/no-gravity-company/no-gravity-elements/commit/2c7e3f0685bab7182cbb76a14e1108d80c4ded93))
