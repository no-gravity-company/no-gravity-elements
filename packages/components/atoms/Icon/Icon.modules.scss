@import '../../../stylesheets/base/variables';

:host {
  display: contents;
  span {
    display: block;

    svg {
      display: block;
      width: 100%;
      height: 100%;
    }

    svg:not([fill='none']),
    svg [fill]:not([fill='none']) {
      fill: var(--text-color, #{$primary}) !important;
      stroke: var(--text-color, #{$primary}) !important;
    }

    svg path[stroke] {
      stroke: var(--text-color, #{$primary}) !important;
    }

    &.extra-small {
      width: baseUnit(1.5);
    }

    &.small {
      width: baseUnit(2);
    }

    &.medium {
      width: baseUnit(2.5);
    }

    &.regular {
      width: baseUnit(3);
    }

    &.large {
      width: baseUnit(4);
    }

    &.extra-large {
      width: baseUnit(6);
    }
  }
}
