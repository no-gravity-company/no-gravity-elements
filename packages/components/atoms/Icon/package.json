{
  "name": "@no-gravity-elements/icon",
  "version": "1.3.4",
  "description": "No Gravity Element",
  "author": "No Gravity Company",
  "private": false,
  "homepage": "https://github.com/no-gravity-company/no-gravity-elements#readme",
  "license": "ISC",
  "main": "lib/index.js",
  "typings": "types/index.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib/index.js",
    "lib/index.js.map",
    "types/index.ts"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/no-gravity-company/no-gravity-elements.git"
  },
  "scripts": {
    "publish-component": "yarn publish --access=public"
  },
  "bugs": {
    "url": "https://github.com/no-gravity-company/no-gravity-elements/issues"
  }
}
