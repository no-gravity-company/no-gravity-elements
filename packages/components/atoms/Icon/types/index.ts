import { IconNames, IconSizes } from '@types';

export interface IconProps {
  name: IconNames;
  size?: IconSizes;
}
