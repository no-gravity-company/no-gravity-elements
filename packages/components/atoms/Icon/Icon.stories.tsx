import { Meta, StoryObj } from '@storybook/web-components';
import { html } from 'lit-html';

import { IconProps } from '@atoms/Icon/types';

import { IconNames, IconSizes } from '@types';

import '@no-gravity-elements/icon';

type CustomArgs = IconProps & { text: string };

const meta: Meta<CustomArgs> = {
  title: 'atoms/Icon',
  component: 'nge-icon',
  parameters: {
    webComponents: {
      tagName: 'nge-icon',
    },
  },
  argTypes: {
    name: {
      description: 'Icon name',
      options: Object.values(IconNames),
    },
    size: {
      description: 'Icon size',
      options: Object.values(IconSizes),
      table: {
        defaultValue: { summary: IconSizes.regular },
      },
    },
  },
  render: ({ name, size }) => html` <nge-icon name=${name} size=${size}></nge-icon> `,
};

export default meta;
type Story = StoryObj<CustomArgs>;

export const Default: Story = {
  args: {
    size: IconSizes.regular,
    name: IconNames.cross,
  },
};
