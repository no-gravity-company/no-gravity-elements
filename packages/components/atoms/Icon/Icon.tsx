import { Fragment, FunctionComponent, h } from 'preact';

import { IconProps } from '@atoms/Icon/types';

import { useFetchSVG } from '@hooks/useFetchSVG';

import { IconSizes } from '@types';

const Icon: FunctionComponent<IconProps> = ({ name, size }: IconProps) => {
  const svgComponent = useFetchSVG(name);

  return (
    <Fragment>
      {svgComponent && <span className={size || IconSizes.regular}>{svgComponent}</span>}
    </Fragment>
  );
};

Icon.observedAttributes = ['size', 'name'];

export default Icon;
