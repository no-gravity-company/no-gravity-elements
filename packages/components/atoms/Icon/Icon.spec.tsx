import { h } from 'preact';
import { shallow } from 'enzyme';

import Icon from '@atoms/Icon/Icon';

import { IconNames, IconSizes } from '@types';

describe('Icon', () => {
  beforeEach(() => {
    jest.spyOn(global, 'fetch').mockImplementation(() =>
      Promise.resolve({
        ok: true,
        text: () =>
          Promise.resolve(
            '<svg xmlns="http://www.w3.org/2000/svg" id="Outline" viewBox="0 0 24 24" width="512" height="512"><path d="M17.17,24a1,1,0,0,1-.71-.29L8.29,15.54a5,5,0,0,1,0-7.08L16.46.29a1,1,0,1,1,1.42,1.42L9.71,9.88a3,3,0,0,0,0,4.24l8.17,8.17a1,1,0,0,1,0,1.42A1,1,0,0,1,17.17,24Z"/></svg>',
          ),
      } as any),
    );
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });
  it('should match the snapshot', async () => {
    let wrapper;
    await new Promise((resolve) => {
      wrapper = shallow(<Icon name={IconNames.cross} size={IconSizes.regular} />);
      setTimeout(resolve, 0);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it.each`
    name               | size
    ${IconNames.cross} | ${IconSizes.extraLarge}
    ${IconNames.cross} | ${IconSizes.large}
  `('should render correctly with the provided name and size', ({ name, size }) => {
    const wrapper = shallow(<Icon name={name} size={size} />);

    const span = wrapper.find(`span.${size}`);
    const svg = wrapper.find('svg');
    expect(span).not.toBe(null);
    expect(svg).not.toBe(null);
  });
});
