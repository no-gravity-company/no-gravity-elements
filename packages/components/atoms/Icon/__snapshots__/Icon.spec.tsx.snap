// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match the snapshot 1`] = `null`;
